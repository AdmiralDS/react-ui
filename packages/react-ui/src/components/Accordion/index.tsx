import * as React from 'react';
import { keyboardKey } from '#src/components/common/keyboardKey';

import { moveFocus, nextItem, previousItem } from './utils';
import { AccordionWrapper } from './style';

type Dimension = 'l' | 'm';

export * from './AccordionItem';

export interface AccordionProps extends React.HTMLAttributes<HTMLDivElement> {
  /** Размер компонента */
  dimension?: Dimension;
  /** Расположение иконки шеврона в заголовке. По умолчанию иконка выравнивается по правому краю. */
  iconPosition?: 'right' | 'left';
  /** Скрыть верхний разделитель аккордиона (верхнюю серую полоску).
   * Параметр hideTopDivider скрывает верхнюю серую полоску только у первого AccordionItem, заданного внутри Accordion
   */
  hideTopDivider?: boolean;
  /** Скрыть нижний разделитель аккордиона (нижнюю серую полоску)
   * Параметр hideBottomDivider скрывает нижнюю серую полоску только у последнего AccordionItem, заданного внутри Accordion
   */
  hideBottomDivider?: boolean;
}

export const Accordion: React.FC<AccordionProps> = ({
  children,
  dimension = 'l',
  iconPosition = 'right',
  hideTopDivider = false,
  hideBottomDivider = false,
  onKeyDown,
  ...props
}) => {
  const accordionRef = React.useRef<HTMLDivElement | null>(null);
  const handleKeyDown = React.useCallback(
    (e) => {
      let focusedOption: any = ((accordionRef.current && accordionRef.current.ownerDocument) || document).activeElement;
      // если фокус находится на AccordionItem Title, то можно переключаться между accordion items с помощью ArrowDown, ArrowUp, Home, End
      if (focusedOption.id.indexOf('accordion_title') > -1 && focusedOption.tagName.toLowerCase() === 'button') {
        do {
          focusedOption = focusedOption?.parentNode;
        } while (focusedOption.parentNode !== accordionRef.current);

        const code = keyboardKey.getCode(e);

        if (code === keyboardKey.ArrowDown) {
          moveFocus(accordionRef.current, focusedOption, nextItem);
          e.preventDefault();
        } else if (code === keyboardKey.ArrowUp) {
          moveFocus(accordionRef.current, focusedOption, previousItem);
          e.preventDefault();
        } else if (code === keyboardKey.Home) {
          e.preventDefault();
          moveFocus(accordionRef.current, null, nextItem);
        } else if (code === keyboardKey.End) {
          e.preventDefault();
          moveFocus(accordionRef.current, null, previousItem);
        }
      }
      onKeyDown?.(e);
    },
    [accordionRef.current],
  );
  return (
    <AccordionWrapper
      ref={accordionRef}
      data-dimension={dimension}
      data-icon={iconPosition}
      hideTopDivider={hideTopDivider}
      hideBottomDivider={hideBottomDivider}
      dimension={dimension}
      onKeyDown={handleKeyDown}
      {...props}
    >
      {children}
    </AccordionWrapper>
  );
};

Accordion.displayName = 'Accordion';
