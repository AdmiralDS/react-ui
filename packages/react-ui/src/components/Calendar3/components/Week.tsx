import type { FC } from 'react';
import * as React from 'react';
import type { Dayjs } from 'dayjs';
import { ThemeContext } from 'styled-components';
import { LIGHT_THEME } from '#src/components/themes';

import { DAY_NUMBERS } from '../constants';
import { addDays, startOfWeek } from '../date-utils';
import { Day } from './Day';
import { WeekComponent } from '../styled/WeekComponent';
import type { IWeekCalendarProps } from '../interfaces';

export const Week: FC<IWeekCalendarProps> = ({
  day,
  month,
  startDate,
  endDate,
  selected,
  activeDate,
  range,
  validator,
  filterDate,
  onMouseEnter,
  onClick,
  highlightSpecialDay,
}) => {
  const theme = React.useContext(ThemeContext) || LIGHT_THEME;
  const handleMouseEnter = (day: Dayjs, e: any) => onMouseEnter && onMouseEnter(day, e);
  const handleDayClick = (day: Dayjs, e: any) => onClick && onClick(day, e);
  const weekStart = startOfWeek(day, theme.locales[theme.currentLocale].firstDayOfWeek ?? 1);
  return (
    <WeekComponent>
      {DAY_NUMBERS.map((offset) => {
        const nextDay = addDays(weekStart, offset);
        return (
          <Day
            key={nextDay.valueOf()}
            day={nextDay}
            month={month}
            startDate={startDate}
            endDate={endDate}
            selected={selected}
            activeDate={activeDate}
            range={range}
            validator={validator}
            filterDate={filterDate}
            onMouseEnter={(_, e) => handleMouseEnter(nextDay, e)}
            onClick={(_, e) => handleDayClick(nextDay, e)}
            highlightSpecialDay={highlightSpecialDay}
          />
        );
      })}
    </WeekComponent>
  );
};
