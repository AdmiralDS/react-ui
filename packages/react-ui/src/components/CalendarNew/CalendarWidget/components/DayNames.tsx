import type { FC } from 'react';
import * as React from 'react';
import type { Dayjs } from 'dayjs';
import 'dayjs/locale/ru';
import { ThemeContext } from 'styled-components';
import { LIGHT_THEME } from '#src/components/themes';

import { capitalizeFirstLetter, DAY_NUMBERS } from '../constants';
import { addDays } from '../date-utils';
import { DayNameComponent } from '../styled/DayNameComponent';

interface IDayNamesProps {
  date: Dayjs;
  userLocale?: string;
}

export const DayNames = ({ date, userLocale }: IDayNamesProps) => {
  const theme = React.useContext(ThemeContext) || LIGHT_THEME;
  const [currentLocale, setCurrentLocale] = React.useState<string>('');

  const defineLocale = userLocale || theme.currentLocale || 'ru';

  if (currentLocale !== defineLocale) {
    import(`dayjs/locale/${defineLocale}.js`)
      .then(() => setCurrentLocale(defineLocale))
      .catch(() => {
        setCurrentLocale('ru');
      });
  }

  const renderNames = () => {
    const instanceLocaleData = date.locale(currentLocale).localeData();
    const start = instanceLocaleData.firstDayOfWeek();
    const weekStart = date.day(start);

    return (
      <div>
        {DAY_NUMBERS.map((offset) => {
          const day = addDays(weekStart, offset);
          const weekDayName = capitalizeFirstLetter(day.locale(currentLocale).format('dd'));

          return <DayNameComponent key={weekDayName}>{weekDayName}</DayNameComponent>;
        })}
      </div>
    );
  };

  return currentLocale ? renderNames() : null;
};
