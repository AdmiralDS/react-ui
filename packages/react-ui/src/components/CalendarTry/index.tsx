import * as React from 'react';
import type { Dayjs } from 'dayjs';
import dayjs from 'dayjs';
import 'dayjs/locale/ru';
import styled, { ThemeContext } from 'styled-components';
import { LIGHT_THEME } from '#src/components/themes';
import { typography } from '#src/components/Typography';
import { DayCell } from '#src/components/CalendarTry/CalendarContent/DayCell';
import { Panel } from '#src/components/CalendarTry/Panel/Panel';
import type { DateValidator } from '#src/components/CalendarTry/validator';
import { YearCell } from '#src/components/CalendarTry/CalendarContent/YearCell';
import type { CalendarViewMode } from '#src/components/CalendarTry/constants';
import { MonthCell } from '#src/components/CalendarTry/CalendarContent/MonthCell';
import { CalendarContent } from '#src/components/CalendarTry/CalendarContent/CalendarContent';

const CALENDAR_WIDTH = 284;
const YEARS_VIEW_PADDING = '20px 12px 16px';
const MONTHS_VIEW_PADDING = '20px 16px 4px';
const DAYS_VIEW_PADDING = '20px 12px 12px';

export const CalendarWidgetWrapper = styled.div<{ viewMode: CalendarViewMode }>`
  position: relative;
  box-sizing: border-box;
  text-align: center;
  border: 0 none;
  //z-index: 0; /* to fix range rounded corners fill */

  padding: ${(p) =>
    p.viewMode === 'YEARS' ? YEARS_VIEW_PADDING : p.viewMode === 'MONTHS' ? MONTHS_VIEW_PADDING : DAYS_VIEW_PADDING};

  width: ${CALENDAR_WIDTH}px;
  background: ${({ theme }) => theme.color['Special/Elevated BG']};
  ${typography['Body/Body 2 Long']}
  color: ${({ theme }) => theme.color['Neutral/Neutral 90']};
`;

export interface CalendarTryProps {
  viewMode?: CalendarViewMode;
  selected?: Dayjs;
  minDate?: Dayjs;
  maxDate?: Dayjs;
  renderCell?: (date: Dayjs, viewMode: CalendarViewMode) => React.ReactNode;
  validator?: DateValidator;
  onSelectDate?: (date: Dayjs) => void;
  onSelectMonth?: (date: Dayjs) => void;
  onSelectYear?: (date: Dayjs) => void;
  disabledDate?: (date: Dayjs) => boolean;
  isHiddenDate?: (date: Dayjs) => boolean;
  onPanelChange?: (date: Dayjs) => void;
  userLocale?: string;
  locale?: {
    backwardText?: string;
    forwardText?: string;
    nextMonthText?: string;
    previousMonthText?: string;
    returnText?: string;
    selectYearText?: string;
    selectMonthText?: string;
  };
}

export const CalendarTry = React.forwardRef<HTMLDivElement, CalendarTryProps>(
  (
    {
      viewMode = 'DATES',
      selected,
      minDate,
      maxDate,
      renderCell,
      validator,
      disabledDate,
      isHiddenDate,
      userLocale,
      onSelectDate,
      onSelectMonth,
      onSelectYear,
      onPanelChange,
      locale,
    },
    ref,
  ) => {
    const getInitialViewDate = () => {
      const current = dayjs();
      if (selected) {
        return selected.locale(currentLocale || 'ru');
      } else {
        if (minDate && current.isBefore(minDate)) {
          return minDate;
        } else if (maxDate && current.isAfter(maxDate)) {
          return maxDate;
        }
      }
      return current.locale(currentLocale || 'ru');
    };
    const theme = React.useContext(ThemeContext) || LIGHT_THEME;
    const [currentLocale, setCurrentLocale] = React.useState<string>();

    const defineLocale = userLocale || theme.currentLocale;
    if (currentLocale !== defineLocale) {
      import(`dayjs/locale/${defineLocale}.js`)
        .then(() => setCurrentLocale(defineLocale))
        .catch(() => {
          setCurrentLocale('ru');
        });
    }

    const [viewDate, setViewDate] = React.useState(getInitialViewDate());

    React.useEffect(() => {
      if (onPanelChange) {
        onPanelChange(viewDate);
      }
    }, [viewDate]);

    React.useEffect(() => {
      if (currentLocale) {
        setViewDate((date) => {
          return date.locale(currentLocale);
        });
      }
    }, [currentLocale]);

    const defaultIsHidden = (date: Dayjs) => {
      return !date.isSame(viewDate, 'month');
    };

    const handleYearClick = (date: Dayjs) => {
      /*changeYear(date.getFullYear());
      !currentActiveViewImportant && setYearsView(false);*/
      onSelectYear && onSelectYear(date);
    };

    const handleMonthClick = (date: Dayjs) => {
      /*changeMonth(date.getMonth());
      !currentActiveViewImportant && setMonthsView(false);*/
      onSelectMonth && onSelectMonth(date);
    };

    const defaultRenderCell = (date: Dayjs, viewMode: CalendarViewMode) => {
      if (viewMode === 'DATES') {
        return (
          <DayCell
            key={date.valueOf()}
            date={date}
            selected={selected}
            disabled={disabledDate?.(date)}
            onSelectDate={onSelectDate}
            isHidden={isHiddenDate?.(date) || defaultIsHidden(date)}
          />
        );
      }
      if (viewMode === 'MONTHS') {
        return (
          <MonthCell
            key={date.valueOf()}
            date={date}
            selected={selected}
            validator={validator}
            onSelectMonth={handleMonthClick}
          />
        );
      }
      if (viewMode === 'YEARS') {
        return (
          <YearCell
            key={date.valueOf()}
            date={date}
            selected={selected}
            validator={validator}
            onSelectYear={handleYearClick}
          />
        );
      }
    };

    const increaseMonth = () =>
      setViewDate((date) => {
        const increase = date.add(1, 'month');
        //onDateIncreaseDecrease && onDateIncreaseDecrease(increase);
        return increase;
      });
    const decreaseMonth = () =>
      setViewDate((date) => {
        const decrease = date.subtract(1, 'month');
        //onDateIncreaseDecrease && onDateIncreaseDecrease(decrease);
        return decrease;
      });

    const renderPanel = () => {
      return (
        <Panel
          viewMode={viewMode}
          date={viewDate}
          userLocale={currentLocale}
          locale={locale}
          onNext={increaseMonth}
          onPrevious={decreaseMonth}
        />
      );
    };

    return currentLocale ? (
      <CalendarWidgetWrapper ref={ref} viewMode={viewMode}>
        {renderPanel()}
        <CalendarContent viewMode={viewMode} date={viewDate} renderCell={renderCell || defaultRenderCell} />
      </CalendarWidgetWrapper>
    ) : null;
  },
);
