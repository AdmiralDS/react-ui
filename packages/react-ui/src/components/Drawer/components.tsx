import * as React from 'react';
import styled from 'styled-components';
import { typography } from '#src/components/Typography';
import { checkOverflow } from '#src/components/common/utils/checkOverflow';

const Title = styled.h5<{ mobile: boolean; displayCloseIcon: boolean }>`
  ${({ mobile }) => (mobile ? typography['Header/H6'] : typography['Header/H5'])};
  color: ${({ theme }) => theme.color['Neutral/Neutral 90']};
  margin: 0;
  padding: ${({ mobile, displayCloseIcon }) => {
    if (mobile) {
      return displayCloseIcon ? '0 48px 8px 16px' : '0 16px 8px';
    }
    return displayCloseIcon ? '0 56px 8px 24px' : '0 24px 8px';
  }};
`;

const Content = styled.div<{ scrollbar: number; mobile: boolean }>`
  overflow-y: auto;
  outline: none;
  padding: ${({ scrollbar, mobile }) => `8px ${(mobile ? 16 : 24) - scrollbar}px 8px ${mobile ? 16 : 24}px`};
  box-sizing: border-box;
`;

const ButtonPanel = styled.div<{ mobile: boolean }>`
  display: flex;
  flex-direction: ${({ mobile }) => (mobile ? 'column-reverse' : 'row-reverse')};
  padding: ${({ mobile }) => (mobile ? '16px 16px 0' : '16px 24px 0')};

  & > button {
    margin: ${({ mobile }) => (mobile ? '0 0 16px 0' : '0 16px 0 0')};
    ${({ mobile }) => mobile && 'width: 100%;'}
  }

  & > button:first-child {
    margin: 0;
  }
`;

export const DrawerContext = React.createContext({ mobile: false, displayCloseIcon: true } as {
  mobile: boolean;
  displayCloseIcon: boolean;
});

export const DrawerTitle: React.FC<React.HTMLAttributes<HTMLHeadingElement>> = ({ children, ...props }) => {
  const { mobile, displayCloseIcon } = React.useContext(DrawerContext);
  const asProp = mobile ? 'h6' : 'h5';
  return (
    <Title mobile={mobile} displayCloseIcon={displayCloseIcon} as={asProp} {...props}>
      {children}
    </Title>
  );
};

export const DrawerContent: React.FC<React.HTMLAttributes<HTMLDivElement>> = ({ children, ...props }) => {
  const contentRef = React.useRef<HTMLDivElement | null>(null);
  const [overflow, setOverflow] = React.useState(false);
  const [scrollbarSize, setScrollbarSize] = React.useState(0);
  const mobile = React.useContext(DrawerContext).mobile;

  React.useLayoutEffect(() => {
    if (contentRef.current && checkOverflow(contentRef.current) !== overflow) {
      setScrollbarSize(contentRef.current.offsetWidth - contentRef.current.clientWidth);
      setOverflow(checkOverflow(contentRef.current));
    }
  }, [children, overflow, setOverflow]);

  React.useLayoutEffect(() => {
    if (contentRef.current) {
      const resizeObserver = new ResizeObserver(() => {
        if (contentRef.current && checkOverflow(contentRef.current) !== overflow) {
          setScrollbarSize(contentRef.current.offsetWidth - contentRef.current.clientWidth);
          setOverflow(checkOverflow(contentRef.current));
        }
      });
      resizeObserver.observe(contentRef.current);
      return () => {
        resizeObserver.disconnect();
      };
    }
  }, [overflow, setOverflow]);

  return (
    <Content tabIndex={-1} ref={contentRef} scrollbar={overflow ? scrollbarSize : 0} mobile={mobile} {...props}>
      {children}
    </Content>
  );
};

export const DrawerButtonPanel: React.FC<React.HTMLAttributes<HTMLDivElement>> = ({ children, ...props }) => {
  const mobile = React.useContext(DrawerContext).mobile;
  return (
    <ButtonPanel mobile={mobile} {...props}>
      {children}
    </ButtonPanel>
  );
};
