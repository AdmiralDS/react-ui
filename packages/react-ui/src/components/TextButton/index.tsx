import * as React from 'react';
import type { ButtonHTMLAttributes, ReactNode } from 'react';
import styled from 'styled-components';
import { Spinner } from '#src/components/Spinner';
import type { Appearance, Dimension, StyledButtonProps } from './types';
import { appearanceMixin } from './appearanceMixin';
import { ButtonContainer, IconContainer } from '#src/components/TextButton/commonMixin';
import { dimensionMixin } from '#src/components/TextButton/dimensionMixin';
import { skeletonAnimationMixin } from '#src/components/skeleton/animation';

const StyledSpinner = styled(Spinner)`
  position: absolute;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
`;

const StyledButton = styled.button.attrs<StyledButtonProps>((props) => ({
  'data-dimension': props.dimension,
  'data-appearance': props.appearance,
}))<StyledButtonProps>`
  position: relative;
  box-sizing: border-box;
  display: inline-flex;
  flex-direction: ${(p) => (p.displayRight ? 'row-reverse' : 'row')};
  align-items: center;
  vertical-align: middle;
  appearance: none;
  border: none;
  background: transparent;
  padding: 0;
  gap: 8px;
  white-space: nowrap;
  ${(p) => p.skeleton && skeletonAnimationMixin};

  ${ButtonContainer} {
    visibility: ${(p) => (p.skeleton || p.$loading ? 'hidden' : 'visible')};
  }

  ${IconContainer} {
    visibility: ${(p) => (p.skeleton || p.$loading ? 'hidden' : 'visible')};
  }

  ${appearanceMixin};

  ${dimensionMixin};

  pointer-events: ${(p) => (p.$loading || p.skeleton || p.disabled ? 'none' : 'all')};
`;

export interface TextButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  /** Внешний вид кнопки */
  appearance?: Appearance;
  /** Размер кнопки */
  dimension?: Dimension;
  /** Иконка кнопки */
  icon?: ReactNode;
  /** Текст кнопки */
  text?: string;
  /** Позиционирование иконки. По умолчанию - слева */
  displayRight?: boolean;
  /** Состояние загрузки */
  loading?: boolean;
  /** Состояние skeleton */
  skeleton?: boolean;
}

export interface TextButtonInternalProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  /** Для внутреннего использования! Иконка кнопки выпадающего меню */
  statusIcon?: ReactNode;
}

export const TextButton = React.forwardRef<HTMLButtonElement, TextButtonProps & TextButtonInternalProps>(
  (
    {
      appearance = 'primary',
      dimension = 'm',
      icon,
      type = 'button',
      text,
      displayRight = false,
      loading = false,
      skeleton = false,
      statusIcon,
      ...props
    },
    ref,
  ) => {
    return (
      <StyledButton
        {...props}
        ref={ref}
        appearance={appearance}
        dimension={dimension}
        displayRight={displayRight}
        type={type}
        $loading={loading}
        skeleton={skeleton}
      >
        {loading && <StyledSpinner dimension={dimension === 's' ? 'ms' : 'm'} />}
        {icon && <IconContainer>{icon}</IconContainer>}
        <ButtonContainer>
          <span>{text}</span>
        </ButtonContainer>
        {statusIcon && !displayRight && <IconContainer>{statusIcon}</IconContainer>}
      </StyledButton>
    );
  },
);

TextButton.displayName = 'TextButton';
