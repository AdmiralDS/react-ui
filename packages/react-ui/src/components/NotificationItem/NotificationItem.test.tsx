import * as React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { LIGHT_THEME } from '#src/components/themes';
import { TextButton } from '#src/components/TextButton';
import {
  NotificationItem,
  NotificationItemButtonPanel,
  NotificationItemContent,
  NotificationItemTitle,
} from '#src/components/NotificationItem/index';

describe('NotificationItem', () => {
  it('should render error components with all properties', () => {
    const handleClick = jest.fn();
    const wrapper = render(
      <ThemeProvider theme={LIGHT_THEME}>
        <NotificationItem status="info" isClosable displayStatusIcon onClose={handleClick}>
          <NotificationItemTitle>Title</NotificationItemTitle>
          <NotificationItemContent>text</NotificationItemContent>
          <NotificationItemButtonPanel>
            <TextButton dimension="s" text="Link" />
          </NotificationItemButtonPanel>
        </NotificationItem>
      </ThemeProvider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
