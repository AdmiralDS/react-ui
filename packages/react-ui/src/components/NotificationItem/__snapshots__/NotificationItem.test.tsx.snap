// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationItem should render error components with all properties 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c6 {
  position: relative;
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  vertical-align: middle;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: none;
  background: transparent;
  padding: 0;
  white-space: nowrap;
  border-radius: 4px;
  pointer-events: all;
}

.c6[data-appearance='primary'] {
  color: #0062FF;
}

.c6[data-appearance='secondary'] {
  color: #23262D;
}

.c6:focus-visible {
  outline-offset: 2px;
  outline: #0062FF solid 2px;
}

.c6:hover {
  cursor: pointer;
  color: #0046E2;
}

.c6:active {
  color: #0132B0;
}

.c6:disabled {
  cursor: not-allowed;
  color: #B0B5BD;
}

.c6[data-dimension='m'] {
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: 500;
  font-size: 16px;
  line-height: 24px;
  font-feature-settings: 'tnum' on,'lnum' on;
}

.c6[data-dimension='s'] {
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: 500;
  font-size: 14px;
  line-height: 20px;
  font-feature-settings: 'tnum' on,'lnum' on;
}

.c11 {
  position: absolute;
  top: 0;
  left: 0;
  height: 100%;
}

.c11 > svg {
  height: 20px;
  width: 20px;
}

.c10 {
  width: 28px;
  height: 28px;
  border-radius: 50%;
  background-color: transparent;
  pointer-events: none;
  position: absolute;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.c7 {
  position: relative;
  padding: 0;
  margin: 4px;
  box-sizing: border-box;
  border: none;
  background-color: transparent;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  height: 20px;
  width: 20px;
  border-radius: 4px;
  overflow: visible;
  cursor: pointer;
}

.c7 > * {
  pointer-events: none;
}

.c7:disabled {
  cursor: default;
  pointer-events: none;
}

.c7:disabled *[fill^='#'] {
  fill: #B0B5BD;
}

.c7:focus-visible {
  outline-offset: 2px;
  outline: #0062FF solid 2px;
}

.c7:focus > .c9 {
  background-color: transparent;
}

.c7:hover > .c9 {
  background-color: #0000000D;
}

.c7:active > .c9 {
  background-color: #0000001F;
}

.c7:focus-visible > .c9 {
  background-color: transparent;
}

.c0 {
  overflow: hidden;
  position: relative;
  box-sizing: border-box;
  border-radius: 4px;
  padding: 12px 44px 12px 52px;
  border-width: 1px;
  border-style: solid;
  border-color: #0062FF;
  background-color: #EDF5FF;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 2px 0;
}

.c3 {
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: 550;
  font-size: 14px;
  line-height: 20px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #23262D;
  margin-bottom: 4px;
}

.c4 {
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 20px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #23262D;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 4px;
  gap: 16px;
}

.c8 {
  position: absolute;
  top: 10px;
  right: 8px;
}

.c1 {
  position: absolute;
  top: 12px;
  left: 16px;
  width: 24px;
  height: 24px;
}

.c1 svg {
  width: 24px;
  height: 24px;
}

.c1 svg *[fill^='#'] {
  fill: #0062FF;
}

<body>
    <div>
      <div
        aria-live="polite"
        class="c0"
        role="status"
      >
        <div
          class="c1"
        >
          <div />
        </div>
        <div
          class="c2"
        >
          <div
            class="c3"
          >
            Title
          </div>
          <div
            class="c4"
          >
            text
          </div>
          <div
            class="c5"
          >
            <button
              class="c6"
              data-appearance="primary"
              data-dimension="s"
              type="button"
            >
              <div
                class=""
              >
                <span>
                  Link
                </span>
              </div>
            </button>
          </div>
        </div>
        <button
          class="c7 close-button c8"
          type="button"
        >
          <div
            aria-hidden="true"
            class="c9 c10"
          />
          <div
            aria-hidden="true"
            class="c11"
          >
            <div
              aria-hidden="true"
            />
          </div>
        </button>
      </div>
    </div>
  </body>,
  "container": .c6 {
  position: relative;
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  vertical-align: middle;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: none;
  background: transparent;
  padding: 0;
  white-space: nowrap;
  border-radius: 4px;
  pointer-events: all;
}

.c6[data-appearance='primary'] {
  color: #0062FF;
}

.c6[data-appearance='secondary'] {
  color: #23262D;
}

.c6:focus-visible {
  outline-offset: 2px;
  outline: #0062FF solid 2px;
}

.c6:hover {
  cursor: pointer;
  color: #0046E2;
}

.c6:active {
  color: #0132B0;
}

.c6:disabled {
  cursor: not-allowed;
  color: #B0B5BD;
}

.c6[data-dimension='m'] {
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: 500;
  font-size: 16px;
  line-height: 24px;
  font-feature-settings: 'tnum' on,'lnum' on;
}

.c6[data-dimension='s'] {
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: 500;
  font-size: 14px;
  line-height: 20px;
  font-feature-settings: 'tnum' on,'lnum' on;
}

.c11 {
  position: absolute;
  top: 0;
  left: 0;
  height: 100%;
}

.c11 > svg {
  height: 20px;
  width: 20px;
}

.c10 {
  width: 28px;
  height: 28px;
  border-radius: 50%;
  background-color: transparent;
  pointer-events: none;
  position: absolute;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.c7 {
  position: relative;
  padding: 0;
  margin: 4px;
  box-sizing: border-box;
  border: none;
  background-color: transparent;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  height: 20px;
  width: 20px;
  border-radius: 4px;
  overflow: visible;
  cursor: pointer;
}

.c7 > * {
  pointer-events: none;
}

.c7:disabled {
  cursor: default;
  pointer-events: none;
}

.c7:disabled *[fill^='#'] {
  fill: #B0B5BD;
}

.c7:focus-visible {
  outline-offset: 2px;
  outline: #0062FF solid 2px;
}

.c7:focus > .c9 {
  background-color: transparent;
}

.c7:hover > .c9 {
  background-color: #0000000D;
}

.c7:active > .c9 {
  background-color: #0000001F;
}

.c7:focus-visible > .c9 {
  background-color: transparent;
}

.c0 {
  overflow: hidden;
  position: relative;
  box-sizing: border-box;
  border-radius: 4px;
  padding: 12px 44px 12px 52px;
  border-width: 1px;
  border-style: solid;
  border-color: #0062FF;
  background-color: #EDF5FF;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 2px 0;
}

.c3 {
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: 550;
  font-size: 14px;
  line-height: 20px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #23262D;
  margin-bottom: 4px;
}

.c4 {
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 20px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #23262D;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 4px;
  gap: 16px;
}

.c8 {
  position: absolute;
  top: 10px;
  right: 8px;
}

.c1 {
  position: absolute;
  top: 12px;
  left: 16px;
  width: 24px;
  height: 24px;
}

.c1 svg {
  width: 24px;
  height: 24px;
}

.c1 svg *[fill^='#'] {
  fill: #0062FF;
}

<div>
    <div
      aria-live="polite"
      class="c0"
      role="status"
    >
      <div
        class="c1"
      >
        <div />
      </div>
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          Title
        </div>
        <div
          class="c4"
        >
          text
        </div>
        <div
          class="c5"
        >
          <button
            class="c6"
            data-appearance="primary"
            data-dimension="s"
            type="button"
          >
            <div
              class=""
            >
              <span>
                Link
              </span>
            </div>
          </button>
        </div>
      </div>
      <button
        class="c7 close-button c8"
        type="button"
      >
        <div
          aria-hidden="true"
          class="c9 c10"
        />
        <div
          aria-hidden="true"
          class="c11"
        >
          <div
            aria-hidden="true"
          />
        </div>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
